pkg/builtin/unary/acos.go

```diff
@@ -0,0 +1,213 @@
+			ReturnType: types.T_float64,
+			Fn: func(origVec *vector.Vector, proc *process.Process, _ bool) (*vector.Vector, error) {
+				origVecCol := origVec.Col.([]int16)
+				resultVector, err := process.Get(proc, 8*int64(len(origVecCol)), types.Type{Oid: types.T_float64, Size: 8})
+				if err != nil {
+					return nil, err
+				}
+				results := encoding.DecodeFloat64Slice(resultVector.Data)
+				results = results[:len(origVecCol)]
+				resultVector.Col = results
+				nulls.Set(resultVector.Nsp, origVec.Nsp)
+				vector.SetCol(resultVector, acos.AcosInt16(origVecCol, results))
+				return resultVector, nil
+			},
+		},
+		{ // T_int32
+			Typ:        types.T_int32,
+			ReturnType: types.T_float64,
+			Fn: func(origVec *vector.Vector, proc *process.Process, _ bool) (*vector.Vector, error) {
+				origVecCol := origVec.Col.([]int32)
+				resultVector, err := process.Get(proc, 8*int64(len(origVecCol)), types.Type{Oid: types.T_float64, Size: 8})
+				if err != nil {
+					return nil, err
+				}
+				results := encoding.DecodeFloat64Slice(resultVector.Data)
+				results = results[:len(origVecCol)]
+				resultVector.Col = results
+				nulls.Set(resultVector.Nsp, origVec.Nsp)
+				vector.SetCol(resultVector, acos.AcosInt32(origVecCol, results))
+				return resultVector, nil
+			},
+		},
+		{ // T_int64
+			Typ:        types.T_int64,
+			ReturnType: types.T_float64,
+			Fn: func(origVec *vector.Vector, proc *process.Process, _ bool) (*vector.Vector, error) {
+				origVecCol := origVec.Col.([]int64)
+				resultVector, err := process.Get(proc, 8*int64(len(origVecCol)), types.Type{Oid: types.T_float64, Size: 8})
+				if err != nil {
+					return nil, err
+				}
+				results := encoding.DecodeFloat64Slice(resultVector.Data)
+				results = results[:len(origVecCol)]
+				resultVector.Col = results
+				nulls.Set(resultVector.Nsp, origVec.Nsp)
+				vector.SetCol(resultVector, acos.AcosInt64(origVecCol, results))
+				return resultVector, nil
+			},
+		},
+		{ // T_float32
+			Typ:        types.T_float32,
+			ReturnType: types.T_float64,
+			Fn: func(origVec *vector.Vector, proc *process.Process, _ bool) (*vector.Vector, error) {
+				origVecCol := origVec.Col.([]float32)
+				resultVector, err := process.Get(proc, 8*int64(len(origVecCol)), types.Type{Oid: types.T_float64, Size: 8})
+				if err != nil {
+					return nil, err
+				}
+				results := encoding.DecodeFloat64Slice(resultVector.Data)
+				results = results[:len(origVecCol)]
+				resultVector.Col = results
+				nulls.Set(resultVector.Nsp, origVec.Nsp)
+				vector.SetCol(resultVector, acos.AcosFloat32(origVecCol, results))
+				return resultVector, nil
+			},
+		},
+		{ // T_float64
+			Typ:        types.T_float64,
+			ReturnType: types.T_float64,
+			Fn: func(origVec *vector.Vector, proc *process.Process, _ bool) (*vector.Vector, error) {
+				origVecCol := origVec.Col.([]float64)
+				resultVector, err := process.Get(proc, 8*int64(len(origVecCol)), types.Type{Oid: types.T_float64, Size: 8})
```

Comment:
perhaps we could reuse the original vector when possible

Reply:
Thanks for your advice